modules/mkdru_ding/mkdru_ding.module

```diff
@@ -0,0 +1,339 @@
+    }
+    return $a['orderWeight'] < $b['orderWeight'] ? -1 : 1;
+  });
+
+  $settings['facets'] = $values['settings']['facets'];
+  $mkdru_defaults = array(
+    'settings' => $settings,
+    'pz2_path' => $values['pz2_path'],
+    'sp_user' => $values['sp_user'],
+    'sp_pass' => $values['sp_pass'],
+    'sp_server_auth' => $settings['sp_server_auth'],
+    'facets' => $values['settings']['facets'],
+  );
+
+  variable_set('mkdru_defaults', $mkdru_defaults);
+  drupal_set_message(t('The configuration options have been saved.'));
+}
+
+/**
+ * Implements hook_theme().
+ */
+function mkdru_ding_theme() {
+  return array(
+    'mkdru_results' => array(
+      'template' => 'templates/mkdru-results',
+      'arguments' => array(),
+    ),
+    'mkdru_search_form' => array(
+      'template' => 'templates/mkdru-search-form',
+      'render element' => 'form',
+    ),
+    'mkdru_block_facet' => array(
+      'template' => 'templates/mkdru-block-facet',
+      'arguments' => array('class' => NULL),
+    ),
+  );
+}
+
+/**
+ * Custom search form.
+ */
+function mkdru_ding_search_form($form, &$form_state) {
+  $form = array();
+  drupal_add_library('system', 'jquery.bbq');
+
+  $form_state['action'] = "search/meta";
+
+  $form['#attributes']['class'][] = 'search-form';
+  $form['#theme'] = 'mkdru_search_form';
+
+  $params = drupal_get_query_parameters();
+
+  $keys = isset($params['query']) ? $params['query'] : arg(2);
+
+  $form['basic']['keys'] = array(
+    '#type' => 'textfield',
+    '#default_value' => $keys,
+    '#maxlength' => 255,
+    '#attributes' => array(
+      'placeholder' => array(t('Search the catalogues, sources and the website...')),
+    ),
+  );
+
+  $form['basic']['submit'] = array(
+    '#type' => 'submit',
+    '#name' => 'simple_search',
+    '#value' => t('Search'),
+    '#submit' => array('mkdru_ding_search_form_submit'),
+    '#validate' => array('mkdru_ding_search_form_validate'),
+  );
+  return $form;
+}
+
+/**
+ * Validate handler for advanced search form.
+ */
+function mkdru_ding_search_form_validate($form, &$form_state) {
+  $values = $form_state['values'];
+  if (empty($values['keys'])) {
+    form_set_error('keys', t('Please specify at least one term.'));
+  }
+}
+
+/**
+ * Submit handler for simple search form.
+ */
+function mkdru_ding_search_form_submit($form, &$form_state) {
+  $params = array();
+  $values = $form_state['values'];
+  if (!empty($values['keys'])) {
+    $params['query'] = $values['keys'];
+  }
+  $form_state['redirect'] = array(
+    $form_state['action'] . '/' . $form_state['values']['keys'],
+    array(
+      'query' => $params,
+    ),
+  );
+}
+
+/**
+ * Implements hook_search_execute().
+ */
+function mkdru_ding_search_execute($keys = '', $conditions = NULL) {
+  $state = array();
+
+  if ($keys) {
+    $state['query'] = $keys;
+  }
+
+  $mkdru_defaults = variable_get('mkdru_defaults', array());
+  $path = drupal_get_path('module', 'mkdru_ding');
+  theme('mkdru_js', array(
```

Comment:
the output from theme() isn't used... does this actually do anything? also, calling the theme() function directly is strongly discouraged.